server/src/domain/errors/mod.rs

```diff
@@ -0,0 +1,68 @@
+use axum::http::StatusCode;
+use axum::response::{IntoResponse, Response};
+use axum::Json;
+use serde_json::json;
+use thiserror::Error;
+use utoipa::ToSchema;
+
+/// Переназначение Result для ответов хендлеров.
+pub type Result<T, E = AppError> = core::result::Result<T, E>;
+
+/// Ошибки сервиса.
+#[derive(Debug, Error, ToSchema)]
+pub enum AppError {
+    // #[schema(example = "Some error from enum 'Errors'")]
+    // счет не существует
+    #[error("account with name: '{0}' not found")]
+    AccountExistsErr(String),
+    // транзакция не существует
+    #[error("account: '{0}' has no transaction with id: '{1}'")]
+    TransactionExistsErr(String, String),
```

Comment:
```suggestion
    AccountDoesntExistErr(String),
    // транзакция не существует
    #[error("account: '{0}' has no transaction with id: '{1}'")]
    TransactionDoesntExistErr(String, String),
```

Мне кажется, обычно это так делается. А то с первого взгляда я подумал, что это ошибка на случай повторной регистрации аккаунта и т.п.)

Reply:
Done